cmd/cigarToBed/cigarToBed.go

```diff
@@ -0,0 +1,198 @@
+			seqIdx++
+		}
+	}
+	return seqIdx
+}
+
+//raven did not put this helper function into the globalAlignment function because it is used in the globalAlignment function
+//faONe is target, faTwo is query
+func cigarToGraph(target fasta.Fasta, query fasta.Fasta, aln []align.Cigar) *genomeGraph.GenomeGraph {
+	answer := genomeGraph.EmptyGraph()
+	//use targetEnd and queryEnd to track position number for each fasta sequence as we move along.
+	var targetEnd, queryEnd int = 0, 0
+	var curr *genomeGraph.Node
+
+	//Creating the first node. This is done independent of all other number because this node has no 'previous' nodes.  All following code leverages the cigar output (second number printed in the Op position of the struct {}) to determine if the alignment returned a 0:match, 1:insertion, or 2:deletion. All indels are relative to the target sequence.
+	curr, targetEnd, queryEnd = genomeGraph.FaSeqToNode(target, query, targetEnd, queryEnd, aln[0], 0)
+	curr = genomeGraph.AddNode(answer, curr)
+	//Drawing the remaining nodes and all edges. Method for adding edges is based on previous nodes.
+	for i := 1; i < len(aln); i++ {
+		curr, targetEnd, queryEnd = genomeGraph.FaSeqToNode(target, query, targetEnd, queryEnd, aln[i], i)
+		curr = genomeGraph.AddNode(answer, curr)
+		if aln[i].Op == 0 {
+			genomeGraph.AddEdge(&answer.Nodes[i-1], curr, 1)
+			genomeGraph.AddEdge(&answer.Nodes[i-2], curr, 0.5)
+		} else if aln[i].Op == 1 || aln[i].Op == 2 {
+			genomeGraph.AddEdge(&answer.Nodes[i-1], curr, 0.5)
+		} else {
+			log.Fatalf("Error: cigar.Op = %d is unrecognized...\n", aln[i].Op)
+		}
+	}
+	return answer
+}
+
+//raven moved helper functions from main and non-usage functions into this function
+func GlobalAlignment_CigarToBed(inputFileOne *fileio.EasyReader, inputFileTwo *fileio.EasyReader, outFa string, outIns_bed string, outDel_bed string, FirstPos_InsBed int, FirstPos_DelBed int, Chrom string) {
+	//make sure files meet the usage requirements of globalAlignment.go
+	faOne, faDoneOne := fasta.NextFasta(inputFileOne)
+	faTwo, faDoneTwo := fasta.NextFasta(inputFileTwo)
+	fasta.ToUpper(faOne) //raven added these 2 lines to prevent lowercase letters from indexing out of range in align/linearGap.go
+	fasta.ToUpper(faTwo)
+	if faDoneOne || faDoneTwo {
+		log.Fatalf("error, unable to read .fa files, check for > symbol before each name and that each fasta entry is only one line. Check to make sure there are only four bases or N, globalAlignment is unable to use anything outside A-T-G-C-N.\n")
+	}
+	numSeqOne := CountSeqIdx(inputFileOne)
+	numSeqTwo := CountSeqIdx(inputFileTwo)
+	if numSeqOne > 1 || numSeqTwo > 1 {
+		log.Fatalf("multiple sequnces detected in .fa files: %v sequences in the first .fa file and %v sequences in the second .fa file. This program is designed for .fa files with only 1 sequence in them\n", numSeqOne, numSeqTwo)
+	}
+	//attempt at Dan's len(faOne) suggestion
+	//if len(string(faOne)) != 1 || len(string(faTwo)) != 1 {
```

Comment:
So I guess it's fair, that the code at line 71 wouldn't catch if there is no fasta/it is blank. But then again, line 62/63 would fail if there wasn't at least one entry anyways?

Reply:
Looking back to the definition of the fasta.NextFasta function, I agree! Thank you!